5 - Samtools
SAM (Sequence Alignment/Map) format is a generic format for storing large nucleotide sequence alignments. SAM aims to be a format that:

Is flexible enough to store all the alignment information generated by various alignment programs;
Is simple enough to be easily generated by alignment programs or converted from existing alignment formats;
Is compact in file size;
Allows most of operations on the alignment to work on a stream without loading the whole alignment into memory;
Allows the file to be indexed by genomic position to efficiently retrieve all reads aligning to a locus.
SAM Tools provide various utilities for manipulating alignments in the SAM format, including sorting, merging, indexing and generating alignments in a per-position format.

samtools view -bS $sam | samtools sort -o $_sorted.bam $sam
samtools index $index
samtools mpileup -E -uf Ceiba-input-seq.fas  $index > $pileup
bcftools call -cg $pileup > $vcf

view: With no options or regions specified, prints all alignments in the specified input alignment file (in SAM, BAM, or CRAM format) to standard output in SAM format (with no header).
-b Output in the BAM format.
-S Ignored for compatibility with previous samtools versions. Previously this option was required if input was in SAM format, but now the correct format is automatically detected by examining the first few characters of input.

sort: Sort alignments by leftmost coordinates, or by read name when -n is used. An appropriate @HD-SO sort order header tag will be added or an existing one updated if necessary.  
-o FILE Write the final sorted output to FILE, rather than to standard output.

index: Index a coordinate-sorted BAM or CRAM file for fast random access. (Note that this does not work with SAM files even if they are bgzip compressed â€” to index such files, use tabix(1) instead.)

mpileup: Generate VCF, BCF or pileup for one or multiple BAM files. Alignment records are grouped by sample (SM) identifiers in @RG header lines. If sample identifiers are absent, each input file is regarded as one sample.
-E, --redo-BAQ Recalculate BAQ on the fly, ignore existing BQ tags
-uF -f, --fasta-ref FILE
The faidx-indexed reference file in the FASTA format. The file can be optionally compressed by bgzip. [null]